dss-api.yml

```diff
@@ -1285,6 +1529,52 @@ definitions:
       total_hits:
         description: The total number of matching results found.
         type: integer
+  Collection:
+    type: object
+    properties:
+      name:
+        description: A short name identifying the collection.
+        type: string
+      description:
+        description: A long description of the collection, formatted in Markdown.
+        type: string
+      details:
+        description: Supplementary JSON metadata for the collection.
+        type: object
+      contents:
+        description: |
+          A list of objects describing links to files, bundles, other collections, and metadata fragments that are
+          part of the collection.
+        type: array
+        items:
+          $ref: '#/definitions/CollectionItem'
+    required:
+      - name
+      - description
+      - details
+      - contents
+  CollectionItem:
+    type: object
+    properties:
+      type:
+        type: string
```

Comment:
should this be an enum?  i would argue yes.

Reply:
Also, what's the magic other type?  I don't get it.  Why not just make it explicit?

Reply:
Enum does not allow a wildcard. I specifically wanted custom types to be declarable. This allows the collection pointer schema to be extensible from upstream. For example, imagine someone wants to express "species" as a first class collection content type (made up example). They could declare a type "species" and use the JSON Pointer to point to a particular fragment of a metadata file representing a species in their metadata schema. The intent is to allow entities that are not first class in the DSS API to be consistently referenced within DSS.

Reply:
The point of this design is to allow custom types. I don't see the utility of controlling the namespace. I suppose we could tighten this up so type identifiers use only alphanumeric characters/underscores etc. I suppose we could also reserve a prefix (e.g. no custom types that start with "dss"), but at this stage my design is deliberately loosely defined to see how it can best accommodate metadata contents/patterns in releases.